test/test_puma_server.rb

```diff
@@ -159,6 +159,32 @@ def test_GET_with_empty_body_has_sane_chunking
     assert_equal "HTTP/1.0 200 OK\r\nContent-Length: 0\r\n\r\n", data
   end
 
+  def test_early_hints_works
+    @server.app = proc { |env|
+     env['rack.early_hints'].call([{ link: "/style.css", as: "style" }, { link: "/script.js" }])
+     [200, { "X-Hello" => "World" }, ["Hello world!"]]
+    }
+
+    @server.add_tcp_listener @host, @port
+    @server.early_hints!
+    @server.run
+
+    sock = TCPSocket.new @host, @server.connected_port
+    sock << "HEAD / HTTP/1.0\r\n\r\n"
+
+    data = sock.read
+
+    assert_equal <<~eos.split("\n").join("\r\n") + "\r\n\r\n", data
```

Comment:
Rubocop is barfing on your squiggly heredoc.

Reply:
I saw the error, and I think RuboCop needs to be fixed.  This is new syntax in Ruby 2.3.  But also, since this is new syntax in Ruby 2.3, and you probably want to run the Puma tests on older versions of Ruby anyway, we'll fix it. 😉

Reply:
Fixed the tests here too. :+1: I also added a test to test that early hints is off by default.